.github/workflows/deploy.yml

```diff
@@ -0,0 +1,149 @@
+      build-enable-npm-caching:
+        required: false
+        type: string
+        default: "false"
+        description: >
+          If set to "true", the NPM packages will be cached by GitHub. This is faster if you have a lot of NPM packages,
+          but negligible or even slower if you only have a handful.
+      build-cache-version:
+        required: false
+        type: string
+        default: "0"
+        description: >
+          Change this to any other value to alter the cache key, effectively invalidating the latest cache. This is the
+          easiest current way to force "clear" the cache
+          (https://github.community/t/how-to-clear-cache-in-github-actions/129038/5) until the associated issue
+          (https://github.com/actions/cache/issues/2) is resolved and the feature is available not just via the RESTful
+          API.
+      timeout-minutes:
+        required: false
+        type: number
+        default: 360
+        description: Configuration for the timeout-minutes parameter of the workflow. The 360 is GitHub's default.
+      app-name:
+        required: true
+        type: string
+        description: The name of the existing app service.
+      slot-name:
+        required: true
+        type: string
+        description: Name of the web app slot.
+      resource-group-name:
+        required: true
+        type: string
+        description: Name of the resource group.
+      url:
+        required: true
+        type: string
+        description: The URL of the web app slot.
+
+jobs:
+  deploy:
+    runs-on: ${{ matrix.machine-type }}
+    name: Deploy to Azure App Service
+    # Timeout-minutes is in the matrix only because the config can't be set directly from the inputs, see:
+    # https://github.com/actions/runner/issues/1555.
+    strategy:
+      matrix:
+        machine-type: ${{ fromJson(inputs.machine-types) }}
+        timeout-minutes:
+        - ${{ inputs.timeout-minutes }}
+    timeout-minutes: ${{ matrix.timeout-minutes }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+
+    - name: Set up .NET
+      uses: Lombiq/GitHub-Actions/.github/actions/setup-dotnet@dev
+      with:
+        dotnet-version: ${{ inputs.dotnet-version }}
+
+    - name: Enable Node corepack
+      uses: Lombiq/GitHub-Actions/.github/actions/enable-corepack@dev
+
+    - name: Build and Static Code Analysis
+      uses: Lombiq/GitHub-Actions/.github/actions/build-dotnet@dev
+      with:
+        directory: ${{ inputs.build-directory }}
+        verbosity: ${{ inputs.build-verbosity }}
+        enable-code-analysis: ${{ inputs.build-enable-code-analysis }}
+        enable-nuget-caching: ${{ inputs.build-enable-nuget-caching }}
+        enable-npm-caching: ${{ inputs.build-enable-npm-caching }}
+        cache-version: ${{ inputs.build-cache-version }}
+
+    - name: Publish
+      run: |
+        dotnet publish -c Release -o '${{ inputs.build-directory }}/Published' 
```

Comment:
Use [`--no-build`](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#:~:text=for%20each%20manifest.-,%2D%2Dno%2Dbuild,-Doesn%27t%20build%20the). Also, for CLI parameters always use their full names so the command is better readable (like `--configuration` instead of `-c`).